### PR TITLE
set IS_IN_CLUSTER on couchbase primary node, fix problem noticed in #427

### PR DESCRIPTION
**Pull request by ahgittin**

### CODE REVIEW

File: software/nosql/src/main/java/brooklyn/entity/nosql/couchbase/CouchbaseClusterImpl.java [hunk 1]
```diff
@@ -229,8 +229,13 @@ protected void doStart() {
...
                 
                 ((CouchbaseNode)getPrimaryNode()).rebalance();
-            } else if (getQuorumSize()>1) {
-                log.warn(this+" is not quorate; will likely fail later, but proceeding for now");
+            } else {
+                if (getQuorumSize()>1) {
+                    log.warn(this+" is not quorate; will likely fail later, but proceeding for now");
+                }
+                if (!serversToAdd.isEmpty()) {
+                    ((EntityInternal) Iterables.getOnlyElement(serversToAdd)).setAttribute(CouchbaseNode.IS_IN_CLUSTER, true);
```
Comment (aledsage) [inline software/nosql/src/main/java/brooklyn/entity/nosql/couchbase/CouchbaseClusterImpl.java]: How do you know that `serversToAdd` will only have one element? If quorum size was 3, then presumably you could get into this code path with serversToAdd having size 2?

**Comment by aledsage** [comment]
@ahgittin Not sure what the failing build was from - it's too old so has been deleted from apache history. 

Can you address the comment about `getOnlyElement` and thus force the build to be kicked off again? (or if my comment is wrong, then get another build by closing+reopening or whatever).

**Comment by ahgittin** [comment]
@aledsage I think you're right, have fixed

**Comment by aledsage** [comment]
Last test failure looks unrelated:

    io.brooklyn.camp.rest.resource.PlatformRestResourceTest.startServer (from TestSuite)
    java.lang.RuntimeException: unable to find a free port at or above 8080
        at brooklyn.util.net.Networking.nextAvailablePort(Networking.java:156)
        at io.brooklyn.camp.CampServer$CampServerUtils.startServer(CampServer.java:151)

We've had a few failures for trying to get an available port. Maybe we need to mark those as integration tests :-(

File: software/nosql/src/main/java/brooklyn/entity/nosql/couchbase/CouchbaseClusterImpl.java [hunk 2]
```diff
@@ -229,8 +229,13 @@ protected void doStart() {
                 }
                 
                 ((CouchbaseNode)getPrimaryNode()).rebalance();
-            } else if (getQuorumSize()>1) {
-                log.warn(this+" is not quorate; will likely fail later, but proceeding for now");
+            } else {
+                if (getQuorumSize()>1) {
```
Comment (aledsage) [inline software/nosql/src/main/java/brooklyn/entity/nosql/couchbase/CouchbaseClusterImpl.java]: Should this really be `if (getQuorumSize() > serversToAdd.size())`? I won't let that block a merge of the PR though.

**Comment by aledsage** [comment]
Looks good; merging.

File: software/nosql/src/main/java/brooklyn/entity/nosql/couchbase/CouchbaseClusterImpl.java (hunk 2 above)
Comment (ahgittin) [inline software/nosql/src/main/java/brooklyn/entity/nosql/couchbase/CouchbaseClusterImpl.java]: it's long-winded but i think logic is correct.  we've established there are `upNodes` aka `serversToAdd`, and if we are in this `else` block then also either we're not quorate or we are only adding a single server.  in the latter case we might be quorate hence this check before warning.

**Pull request merged**